lib/tasks/server/middleware/proxy-server/index.js

```diff
@@ -26,9 +26,13 @@ class ProxyServerAddon {
         timeout: options.proxyInTimeout,
       });
 
-      proxy.on('error', (e) => {
+      proxy.on('error', (e, req, res) => {
         options.ui.writeLine(`Error proxying to ${options.proxy}`);
         options.ui.writeError(e);
+        if (typeof res.status === 'function') {
```

Comment:
`res` can be a http response or websocket. node-http-proxy emits an error for both:

https://github.com/http-party/node-http-proxy/blob/9b96cd725127a024dabebec6c7ea8c807272223d/lib/http-proxy/passes/web-incoming.js#L165

https://github.com/http-party/node-http-proxy/blob/9b96cd725127a024dabebec6c7ea8c807272223d/lib/http-proxy/passes/ws-incoming.js#L157

node-http-proxy cleans up websockets automatically. We need to explicitly end http requests or they will hang indefinitely in the browser.